gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/FakeStorageRpc.java

```diff
@@ -0,0 +1,264 @@
+package com.google.gcloud.storage.testing;
+
+import com.google.api.services.storage.model.Bucket;
+import com.google.api.services.storage.model.StorageObject;
+import com.google.gcloud.spi.ServiceRpcFactory;
+import com.google.gcloud.spi.StorageRpc;
+import com.google.gcloud.storage.StorageException;
+import com.google.gcloud.storage.StorageOptions;
+
+import sun.misc.IOUtils;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.NoSuchFileException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A bare-bones in-memory implementation of Storage, meant for testing.
+ */
+public class FakeStorageRpc implements StorageRpc {
```

Comment:
I think this class is nice as is, however I think we should move the resetAndGetFakeStorageOptions to a seperate
class named LocalGcsHelper following similar methods as other helpers such as `create` and `options` (so usage-examples would be similar).

Also, we should add a package-info.java with how to use the helper in tests (and any disclaimers, such as not all features are supported,...)